CMakeLists.txt

```diff
@@ -2,6 +2,25 @@ cmake_minimum_required(VERSION 3.12)
 
 project(f3d-superbuild)
 
+macro (superbuild_setup_variables)
+  include(SuperbuildVersionMacros)
+  if (f3d_SOURCE_SELECTION STREQUAL "git")
+    # Assuming master; just use the latest version, but let the user set their
+    # own version in case it is actually a branch from some other version.
+    set(F3D_VERSION_DEFAULT "1.3.1"
+      CACHE STRING "The default version of F3D to use if it cannot be detected")
+    mark_as_advanced(F3D_VERSION_DEFAULT)
+    set(f3d_default_version "${F3D_VERSION_DEFAULT}")
+  elseif (f3d_SOURCE_SELECTION STREQUAL "source")
+    # If it is a git repo, we'll get it that way.
+    set(f3d_default_version "")
+  else ()
+    # The selection is the version number; use it.
+    set(f3d_default_version "${f3d_SOURCE_SELECTION}")
+  endif()
+  superbuild_set_version_variables(f3d "${f3d_default_version}" "f3d-version.cmake")
+endmacro ()
```

Comment:
@Meakk : So this is the version logic of ParaView. The idea is that the superbuild needs to know the version of ParaView before having access to the paraview sources, hence the "_VERSION_DEFAULT" which is not recovered from ParaView but set in hard here.

ParaView needs that but F3D does not, so we could consider changing this logic and only recover the version when we package F3D. This is probably doable by reading the f3dConfigVersion.cmake in the install tree by storing the version in a `version.txt` would make it simpler and more straightforward to recover.

That being said, current approach works fine, but require to duplicate the version number.

wdyt ?

Reply:
As we discussed offline, I think this current approach is fine. Duplication of the release number isn't a big deal.